spec/features/h2_object_creation_spec.rb

```diff
@@ -116,24 +116,11 @@
 
     # Opens Argo detail page
     visit "#{Settings.argo_url}/view/#{bare_druid}"
-    reload_page_until_timeout!(check_wf_error: false) do
-      # NOTE: This is here to work around a persistent, not easily
-      #       reproducible race condition that is occasionally seen in Argo,
-      #       causing integration tests to fail. This work-around mimics the
-      #       steps that developers tend to perform manually.
-      if page.has_text?('Error: shelve : problem with shelve', wait: 0)
-        click_link 'accessionWF'
-        select 'Rerun', from: 'status'
-        confirm_message = 'You have selected to manually change the status. '
-        confirm_message += 'This could result in processing errors. Are you sure you want to proceed?'
-        accept_confirm(confirm_message) do
-          click_button 'Save'
-        end
-        sleep 3
-      end
-
-      page.has_text?('v2 Accessioned', wait: 3)
-    end
+    # wait for accessioningWF to finish; retry if error on shelving step, likely caused by a race condition
+    reload_page_until_timeout_with_wf_step_retry!(expected_text: 'v2 Accessioned',
+                                                  workflow: 'accessionWF',
+                                                  workflow_retry_text: 'Error: shelve : problem with shelve',
+                                                  retry_wait: 10)
```

Comment:
this is the same functionality but more re-usable and clearer (IMO)

Reply:
Note that this needs a much longer wait after the retry of the step than the other test using this method.